src/layout/logfmt.rs

```diff
@@ -69,11 +69,20 @@ impl LogfmtLayout {
     }
 }
 
-/// The idea is borrowed from https://github.com/go-logfmt/logfmt/
-fn escape(s: &str) -> Cow<str> {
-    match s.contains([' ', '=', '"']) {
-        true => format!("\"{}\"", s.escape_debug()).into(),
-        false => s.into(),
+// The encode logic is copied from https://github.com/go-logfmt/logfmt/blob/76262ea7/encode.go.
+fn encode_key_value(result: &mut String, key: &str, value: &str) {
+    use std::fmt::Write;
+
+    if key.contains([' ', '=', '"']) {
+        // omit keys contain special chars
+        return;
+    }
```

Comment:
Maybe return an error as go logfmt does. But it would require diagnostic visitor's visit failable and I'll leave it for a follow-up.